crates/engine/tree/src/tree/mod.rs

```diff
@@ -2347,7 +2347,7 @@ where
         debug!(target: "engine::tree", ?root_elapsed, block=?block_num_hash, "Calculated state root");
 
         // ensure state root matches
-        if state_root != block.header().state_root() {
+        if self.consensus.validate_state_root(block.header(), state_root).is_err() {
```

Comment:
I see there are a couple more state root comparisons (if you search for `if state_root` there are 8 results), do we need to update any of the other ones?